Yarn 2 Dependency Issues

# Bug report

## Describe the bug

There seems to be some dependency issues with v9.3.4 when using Yarn 2.  First one I ran into seems to be here:
https://github.com/zeit/next.js/blob/7f8f149f9986afd8cb916e27eb231bacaad9dc9d/packages/next/build/utils.ts#L6
I was able to work around it by unplugging `next`, and changing that to `next/dist/compiled/strip-ansi`.

But then I started hitting other roadblocks.  For example, `autodll-webpack-plugin` is using `find-cache-dir` (which I noticed is listed as an external), but it is not listed as a dependency for that package.  As a matter of fact, its `package.json` appears to be quite minimal.  It looks to me to be a product of the your build process?  I'll be honest: I didn't take much time to analyze it.  I thought you guys would be better equipped to deal with this.

## To Reproduce

* Clone [my reproduction repository](https://github.com/Js-Brecht/nextjs-yarn2-issue)

* Run `yarn`

* Run `yarn dev`

## Expected behavior

It should start the dev server

## System information

- OS: Linux
- Browser (if applies): Chrome
- Version of Next.js: v9.3.4

## Additional context

v9.3.3 works fine :man_shrugging:

Just to add context, I updated an app to 9.3.4, ran npm run build on it, and ran into this:

```
./node_modules/next/dist/client/next.js
Error: Cannot find module 'find-cache-dir'
Require stack:
```

I installed 'find-cache-dir' in my app's dependencies, and that resolved it.

--- a/packages/next/build/utils.ts
+++ b/packages/next/build/utils.ts
@@ -3,7 +3,7 @@ import gzipSize from 'next/dist/compiled/gzip-size'
 import textTable from 'next/dist/compiled/text-table'
 import path from 'path'
 import { isValidElementType } from 'react-is'
-import stripAnsi from 'strip-ansi'
+import stripAnsi from 'next/dist/compiled/strip-ansi'
 import { Redirect, Rewrite, Header } from '../lib/check-custom-routes'
 import {
   SSG_GET_INITIAL_PROPS_CONFLICT,

